src/main/scala/ai/nikin/typedgraph/core/Edge.scala

```diff
@@ -1,15 +1,31 @@
 package ai.nikin.typedgraph.core
 
+import java.rmi.UnexpectedException
 import scala.annotation.unused
 
 sealed abstract class TypelessEdge(
     private[core] val from: AnyVertex,
     private[core] val to:   AnyVertex,
 ) { self =>
+  override def toString:                 String  = s"$from > $to"
   lazy final private[core] val dropType: AnyEdge = self
   private[core] def asPath:              AnyPath = Path(self)
 
-  private[core] def >?>(next: AnyVertex): AnyPath = asPath >?> next
+  lazy final private[core] val flatten: Set[AnyEdge] =
```

Comment:
Applicable to all uses of Set in the code: this will not guarantee ordering. Can we perhaps use `ListSet`? https://www.scala-lang.org/api/2.13.10/scala/collection/immutable/ListSet.html

Reply:
done